### PR TITLE
Allow `dia_matrix` creation from SciPy equivalent

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/cupy/cupy/issues/3158

### CODE REVIEW

**Comment by jakirkham** [comment]
Have pushed some tests. Would it be possible to get this to run on CI @emcastillo? 🙂 Could also use a review if someone is free 😉

File: cupyx/scipy/sparse/dia.py [hunk 1]
```diff
@@ -33,7 +33,13 @@ class dia_matrix(data._data_matrix):
     format = 'dia'
 
     def __init__(self, arg1, shape=None, dtype=None, copy=False):
-        if isinstance(arg1, tuple):
+        if _scipy_available and scipy.sparse.issparse(arg1):
```
Comment (emcastillo) [inline cupyx/scipy/sparse/dia.py]: Shouldn't check that `arg1` is also stored in dia format?

Comment (jakirkham) [reply cupyx/scipy/sparse/dia.py]: Good point! Thanks 😄

Should be fixed in the latest commit 🙂

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**